homeassistant/components/vevor_heater/__init__.py

```diff
@@ -0,0 +1,66 @@
+"""The Vevor Heater integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothCallbackMatcher,
+    BluetoothScanningMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import VevorHeaterUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Vevor Heater from a config entry."""
+
+    address = entry.unique_id
+    assert address is not None
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, address)
+
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find Vevor Heater device with address {address}"
+        )
+
+    vevor_coordinator = VevorHeaterUpdateCoordinator(hass=hass, device=ble_device)
+
+    await vevor_coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = vevor_coordinator
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    @callback
+    def _async_update_ble(
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        _: bluetooth.BluetoothChange,
+    ) -> None:
+        vevor_coordinator.set_ble_device(service_info.device)
+
+    bluetooth.async_register_callback(
+        hass,
+        _async_update_ble,
+        BluetoothCallbackMatcher(address=address),
+        BluetoothScanningMode.ACTIVE,
+    )
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok and DOMAIN in hass.data:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
```

Comment:
```suggestion
    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok
```

unload will never be called unless its loaded

Reply:
Done.